From 178688683a94dfa480712b787dc910d4ea1ad2da Mon Sep 17 00:00:00 2001
From: Pradeep Sriram <pradeep.sriram@amlogic.com>
Date: Mon, 9 Nov 2020 17:57:51 -0800
Subject: [PATCH] RDK: fix platco issues

PD#SWPL-36538

Problem: compile failed due to header duplicate

Solution: Add linker flags, fix header

Verify: Build pass

Change-Id: I4a69f1152717ec274c2889c861a761a626abad34
Signed-off-by: Pradeep Sriram <pradeep.sriram@amlogic.com>
---
 Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index 612d6cb..86e52a0 100644
--- a/Makefile
+++ b/Makefile
@@ -4,7 +4,7 @@
 #
 ################################################################################
 LOCAL_PATH = $(shell pwd)
-LDFLAGS += -lstdc++ -lpthread -lz -ldl -lrt -L$(STAGING_DIR)/usr/lib
+LDFLAGS += -Wl,--no-as-needed -lstdc++ -lpthread -lz -ldl -lrt -L$(STAGING_DIR)/usr/lib
 CFLAGS += -Wall -Wno-unknown-pragmas -Wno-format \
           -O3 -fexceptions -fnon-call-exceptions -D_GNU_SOURCE \
           -I$(STAGING_DIR)/usr/include -DHAVE_AUDIO
@@ -72,7 +72,7 @@ libtv.so: $(tv_SRCS)
 	-I$(LOCAL_PATH)/libtv/tvutils -laudio_client -o $@ $^ $(LDLIBS)
 
 tvservice: $(tvservice_SRCS) libtv.so
-	$(CC) $(CFLAGS) $(LDFLAGS) -I$(tvclient_HEADERS) \
+	$(CXX) $(CXXFLAGS) $(CFLAGS) $(LDFLAGS) -I$(tvclient_HEADERS) \
 	-I$(LOCAL_PATH)/libtv -I$(LOCAL_PATH)/libtv/tvutils \
 	-L$(LOCAL_PATH) -ltv -laudio_client -o $@ $^ $(LDLIBS)
 
-- 
2.17.1

